src/sentry/search/utils.py

```diff
@@ -215,67 +216,69 @@ def parse_query(project, query, user):
     tokens = tokenize_query(query)
 
     results = {'tags': {}, 'query': []}
-
-    for key, token_list in six.iteritems(tokens):
-        for value in token_list:
-            if key == 'query':
-                results['query'].append(value)
-            elif key == 'is':
-                if value == 'unassigned':
-                    results['unassigned'] = True
-                elif value == 'assigned':
-                    results['unassigned'] = False
-                else:
-                    try:
-                        results['status'] = STATUS_CHOICES[value]
-                    except KeyError:
-                        pass
-            elif key == 'assigned':
-                if value == 'me':
-                    results['assigned_to'] = user
-                else:
-                    try:
-                        results['assigned_to'] = find_users(value)[0]
-                    except IndexError:
-                        # XXX(dcramer): hacky way to avoid showing any results when
-                        # an invalid user is entered
-                        results['assigned_to'] = User(id=0)
-            elif key == 'bookmarks':
-                if value == 'me':
-                    results['bookmarked_by'] = user
-                else:
-                    try:
-                        results['bookmarked_by'] = find_users(value)[0]
-                    except IndexError:
-                        # XXX(dcramer): hacky way to avoid showing any results when
-                        # an invalid user is entered
-                        results['bookmarked_by'] = User(id=0)
-            elif key == 'first-release':
-                results['first_release'] = parse_release(project, value)
-            elif key == 'release':
-                results['tags']['sentry:release'] = parse_release(project, value)
-            elif key == 'user':
-                if ':' in value:
-                    comp, value = value.split(':', 1)
+    try:
```

Comment:
dont mean to be nitpicky, but giant indented block of code is generally frowned upon.

There's a couple of simply things that could be done here:

1. we could make parse_query call something like _parse_query and wrap that in the try/catch
2. we could make parse_release  (or whatever individually items might cause an error) raise InvalidQuery rather than DataError